### ISSUE TITLE
[$500] Android - Create room - Inconsistent behavior of cursor after selecting workspace/visibility

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#31224]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.98.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Open the app
2. Tap the FAB > Start chat > Room
3. Enter the room name
4. Select the workspace and/or visibility

## Expected Result:
The cursor is not shown, and the keyboard is hidden

## Actual Result:
On Android and iOS, the cursor returns to the room name, and the field is highlighted, while on web and mweb, the cursor is not seen, and the keyboard is hidden on mweb. On iOS, the keyboard overlaps the "Create room" button. Also, the keyboard is hidden on Samsung s10+, while it appears on Pixel 6.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/c049a9bf-5ad7-4caf-bed6-42b60096c60a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

https://github.com/Expensify/App/assets/93399543/b058ca62-8f69-4ca9-89c4-d7f83a69cf4c

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01fbab8ee06aa1521a</li>
        <li>Upwork Job ID: 1723173111736639488</li>
        <li>Last Price Increase: 2024-06-15</li>
    </ul>
</details>

**Comment by sakluger** [comment]
I agree with @tienifr that this is an issue and it's not enough to wait for the user to close the modal. Although I don't really understand how this is related to https://github.com/Expensify/App/issues/24452#issuecomment-1712021524, the issue that @bernhardoj references.

@hoangzinh I'm curious for your thoughts here. Do you think we should go with @tienifr's proposal, or would it be better to hold?

**Comment by sakluger** [comment]
Okay, we'll put on hold for https://github.com/Expensify/App/issues/24452 and revisit once that is complete.

**Comment by sakluger** [comment]
Still holding on https://github.com/Expensify/App/issues/24452.

**Comment by sakluger** [comment]
Still holding.

**Comment by sakluger** [comment]
Still holding for https://github.com/Expensify/App/issues/24452.

**Comment by sakluger** [comment]
Still on hold.

**Comment by sakluger** [comment]
Still on hold, the PR for https://github.com/Expensify/App/issues/24452 is in review.

**Comment by sakluger** [comment]
Still on hold.

**Comment by sakluger** [comment]
Still on hold

**Comment by sakluger** [comment]
Still on hold for https://github.com/Expensify/App/issues/24452

**Comment by sakluger** [comment]
Still on hold.

**Comment by sakluger** [comment]
Thanks @hoangzinh. I switched to Monthly, and also added to the vip-vsb project since it has to do with chat functionality.

**Comment by sakluger** [comment]
Still on hold for https://github.com/Expensify/App/issues/24452.

**Comment by sakluger** [comment]
The solution for the hold issue was deployed to prod last week, but I can still reproduce the issue on iOS v1.4.80-17.

@hoangzinh should we open this issue back up for new proposals?

**Comment by bernhardoj** [comment]
I think https://github.com/Expensify/App/issues/24452 is not complete yet, they are splitting the solution into several PRs.

cc: @ntdiary

**Comment by sakluger** [comment]
Thanks @bernhardoj. I asked in that issue too, but do you know which PRs are still pending?

**Comment by bernhardoj** [comment]
I don't see any open PR, but based on this [comment](https://github.com/Expensify/App/issues/24452#issuecomment-1921838876), they are trying to split it into 4 phases and the 2nd phase is already completed.

**Comment by sakluger** [comment]
It sounds like the remaining parts of the hold issue won't solve this, so I've added the help wanted label back.

**Comment by sakluger** [comment]
I'm not an engineer so I can't really evaluate how good that solution is 🙂. @hoangzinh feel free to C+ approve the proposal so we can get a BE engineer assigned to confirm. Thanks!

**Comment by bernhardoj** [comment]
After retesting, I found that the behavior on all platforms now is the same. The room name field is refocused after selecting the workspace visibility because of the focus trap.

**Comment by sakluger** [comment]
Thanks @bernhardoj for retesting! Yes, let's close this out.
